.github/workflows/build-and-publish.yml

```diff
@@ -4,6 +4,9 @@ on:
   push:
     tags:
      - v**
+    branches:
+      - 'upcoming/*'
+      - 'master'
```

Comment:
Are these conditions (tags, branches, pull_requests) ANDed or ORed together?

Reply:
I'm pretty sure these are treated as ORs.

Reply:
So i validated this, they are treated as ORs. Anything tagged with a `v<something>`, pushed to `master`, pushed to `upcoming/<something>` or pull requests against `master` will trigger the workflow.